### PR TITLE
cards component

### PR DESCRIPTION
**Pull request by WhiteLight1107**
### Summary
Terra-card component completes #313 
### Deployment Links 
TBD

### CODE REVIEW

**Comment by emilyrohrbough** [comment]
We won't want to check in the `package-lock.json` file

File: packages/terra-card/docs/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Terra Card
+
+{insert description}
```
Comment (emilyrohrbough) [inline packages/terra-card/docs/README.md]: Add description

File: packages/terra-card/docs/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  - `npm install terra-card`
+  - `yarn add terra-card`
+
+## Usage
+
+```jsx
+import React from 'react';
+import Card from 'terra-card';
+
+<Card {props...} />
```
Comment (emilyrohrbough) [inline packages/terra-card/docs/README.md]: Could you give an example or two of the expected use?

File: packages/terra-card/package.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
+{
+  "name": "terra-card",
+  "main": "lib/Card.js",
+  "version": "0.0.0",
+  "description": "{insert description}",
```
Comment (emilyrohrbough) [inline packages/terra-card/package.json]: Add description

File: packages/terra-card/package.json [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  "author": "Cerner Corporation",
+  "license": "Apache-2.0",
+  "bugs": {
+    "url": "https://github.com/cerner/terra-core/issues"
+  },
+  "homepage": "https://github.com/cerner/terra-core#readme",
+  "peerDependencies": {
+    "react": "^15.4.2",
+    "react-dom": "^15.4.2",
+    "terra-base": "^1.0.0"
```
Comment (emilyrohrbough) [inline packages/terra-card/package.json]: Update to `terra-base: ^1.3.0`

File: packages/terra-card/docs/README.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+```
+
+## Component Features
+
+<!-- Uncomment supported features.
```
Comment (emilyrohrbough) [inline packages/terra-card/docs/README.md]: Uncomment the supported features

File: packages/terra-card/src/Card.jsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import './Card.scss';
+
```
Comment (emilyrohrbough) [inline packages/terra-card/src/Card.jsx]: To create name-spaced class names for css modules, please update
`import './Card.scss';` to `import styles from './Card.scss';`
 and then add
`const cs = classNames.bind(styles);`

To incorporate this style binding, you might need to update how you are applying classNames for vertical and horizontal padding. [Here](https://github.com/cerner/terra-core/blob/master/packages/terra-button/src/Button.jsx) is an example that would be good to look at for guidance.

File: packages/terra-card/src/Card.scss [hunk 7]
```diff
@@ -0,0 +1,27 @@
+
```
Comment (emilyrohrbough) [inline packages/terra-card/src/Card.scss]: For the css module styling, this file needs updated to be locally scoped. [Here](https://github.com/cerner/terra-core/blob/master/packages/terra-button-group/src/ButtonGroup.scss) is a simple example of what your styling might look like. 

Our team has decided to use css styling syntax for class names, so `.cardContainer` should be changed to `.card-container`.

File: packages/terra-card/src/Card.jsx [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+  padHorizontal: PropTypes.bool,
+  /**
+   * Child Nodes
+   */
+  children: PropTypes.node,
+};
+
+const defaultProps = {
+  padVertical: true,
+  padHorizontal: true,
```
Comment (alex-bezek) [inline packages/terra-card/src/Card.jsx]: the defaults should be no padding per this
https://github.com/cerner/terra-core/issues/313#issuecomment-322214889

File: packages/terra-card/README.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
+# Terra Card
+
+
+[![NPM version](http://img.shields.io/npm/v/terra-card.svg)](https://www.npmjs.org/package/terra-card)
+[![Build Status](https://travis-ci.org/cerner/terra-core.svg?branch=master)](https://travis-ci.org/cerner/terra-core)
+
+{insert description}
```
Comment (jmsv6d) [inline packages/terra-card/README.md]: Add description here

File: packages/terra-card/src/Card.jsx [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+};
+
+const Card = ({
+  padVertical,
+  padHorizontal,
+  children,
+  ...customProps
+}) => {
+  // Follows these conventions https://developer.mozilla.org/en-US/docs/Web/CSS/Shorthand_properties
```
Comment (alex-bezek) [inline packages/terra-card/src/Card.jsx]: This comment is no longer applicable as we are just doing the 2 properties instead of the css shorthand style array of values

File: packages/terra-card/src/Card.scss [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+/* stylelint-disable selector-class-pattern  */
+.card {
```
Comment (jmsv6d) [inline packages/terra-card/src/Card.scss]: You should probably discuss with @neilpfeiffer to determine if any of these values should be themeable.

File: packages/terra-card/src/Card.jsx [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+    'card',
+    customProps.className,
+  ]);
+
+  if (padVertical) {
+    containerClass += ' veritcalPadding';
+  }
+  if (padHorizontal) {
+    containerClass += ' horizontalPadding';
+  }
```
Comment (StephenEsser) [inline packages/terra-card/src/Card.jsx]: ClassNames allows conditional class names. This if statement can be replaced with the classNames API.
Example:
```
  const cardClass = cx([
    'card',
    { 'veritcal-padding': padVertical },
    { 'horizontal-padding': padHorizontal },
    customProps.className,
  ]);
```
( Assuming the containers below are flattened into one contianer )

File: packages/terra-card/src/Card.jsx [hunk 13]
```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import './Card.scss';
+
+const propTypes = {
+  /**
+   * Provides 10px on top and bottom
```
Comment (jmsv6d) [inline packages/terra-card/src/Card.jsx]: Is this padding going to be themeable? If so, you should probably reword this to "Provide themeable padding to top and bottom" same with vertical padding.

File: packages/terra-card/tests/nightwatch/card-spec.js [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+
+  afterEach: (browser, done) => {
+    screenshot(browser, 'terra-card', done);
+  },
+
+  'Displays a default card': (browser) => {
+    browser
+      .url(`http://localhost:${browser.globals.webpackDevServerPort}/#/tests/card-tests/default`)
+      .assert.elementPresent('.card');
+  },
```
Comment (jmsv6d) [inline packages/terra-card/tests/nightwatch/card-spec.js]: You aren't testing all of your nightwatch test files.

File: packages/terra-card/src/Card.jsx [hunk 15]
```diff
@@ -0,0 +1,60 @@
...
+  if (padHorizontal) {
+    containerClass += ' horizontalPadding';
+  }
+
+
+  const content = <div className={containerClass}> {children}</div>;
+
+  return (
+    <div {...customProps} className={cardClass}>
+      {content}
```
Comment (StephenEsser) [inline packages/terra-card/src/Card.jsx]: I recommend using `children` here directly, styling the existing outer div, and removing the `content` div wrapper unless the double nested div is needed independent of the wrapper the card class is being applied to.

File: packages/terra-card/src/Card.scss (hunk 11 above)
Comment (StephenEsser) [inline packages/terra-card/src/Card.scss]: +1 I think it is likely the background color will need to be theme-able, which would probably require the box shadow to theme

File: packages/terra-card/src/Card.scss [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+  background-color: #fff;
+  border-radius: 3px;
+  box-shadow: 0 1px 1px 0 #1c1f21;
+  margin: 10px;
```
Comment (StephenEsser) [inline packages/terra-card/src/Card.scss]: Adding on to Jaime's comment above, I would reach out to @neilpfeiffer to see if we want a constant margin around each Card or if the Component the Card is within should control the spacing

File: packages/terra-card/src/Card.jsx (hunk 15 above)
Comment (jmsv6d) [inline packages/terra-card/src/Card.jsx]: +1 to remove nested content div, if possible.

Comment (alex-bezek) [inline packages/terra-card/src/Card.jsx]: The content wrapper div originally was because we were handling clickable cards and it was needed for something. Agreed, let's remove it and can always bring it back later if required for something

File: packages/terra-card/src/Card.scss [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+.cardContainer {
```
Comment (alex-bezek) [inline packages/terra-card/src/Card.scss]: With the removal of the extra div, do we need this as a separate class anymore?

File: packages/terra-card/src/Card.scss [hunk 18]
```diff
@@ -0,0 +1,27 @@
...
+  color: inherit;
+  display: block;
+  height: 100%;
+  text-decoration: none;
```
Comment (alex-bezek) [inline packages/terra-card/src/Card.scss]: I think this was specifically for our clickable card. Is this still required?

File: packages/terra-card/src/Card.scss (hunk 16 above)
Comment (neilpfeiffer) [inline packages/terra-card/src/Card.scss]: I remember from Friday's conversation that there wouldn't be any margin on the card but would be on the responsibility of card-collection or similar grid based parent. Is that a correction assumption of always the containing component handling layout/placement?

It still could be a themeable option, I suppose in addition, but would need to investigate/validate further.

File: packages/terra-card/src/Card.scss (hunk 11 above)
Comment (neilpfeiffer) [inline packages/terra-card/src/Card.scss]: I would say all used style props are probably themeable, my list would include:
border, background, box-shadow, padding (based on relative "sizing set" of css-variables for entire theme)

File: packages/terra-card/src/Card.scss (hunk 18 above)
Comment (neilpfeiffer) [inline packages/terra-card/src/Card.scss]: Ditto - I would be careful of `text-decoration: none` without good reason.

File: packages/terra-card/src/Card.scss (hunk 16 above)
Comment (jmsv6d) [inline packages/terra-card/src/Card.scss]: I would say we should remove the margin for now and a themeable option could be added later if a need for it occurs.

File: packages/terra-card/src/Card.scss (hunk 11 above)
Comment (alex-bezek) [inline packages/terra-card/src/Card.scss]: Just curious, are we going to start making most style's themeable right out of the gate, or add in theme-ability as the need comes? We can always add more customizability, but it's harder to take it away

Comment (mhemesath) [inline packages/terra-card/src/Card.scss]: I'd say we only make theme-able what needs to be. Its fairly trivial to expose additional properties.

File: packages/terra-card/src/Card.jsx (hunk 12 above)
Comment (mhemesath) [inline packages/terra-card/src/Card.jsx]: I'm actually in favor of vertica/horizontal padding being a sub-component (or 2 sub-components). This enables us to reuse those padding styles  in sub ocmponents of the card. Example, we need a side-to-side horizontal rule and want to have 2 padded containment regions above and below it.

Comment (WhiteLight1107) [inline packages/terra-card/src/Card.jsx]: So should I remove the padding properties all together from card? @mhemesath
